tools/analyzer_plugin/test/integration/diagnostics/create_ref_usage_test.dart

```diff
@@ -0,0 +1,143 @@
+// Disable null-safety in the plugin entrypoint until all dependencies are null-safe,
+// otherwise tests won't be able to run. See: https://github.com/dart-lang/test#compiler-flags
+// @dart=2.9
+import 'dart:async';
+
+import 'package:over_react_analyzer_plugin/src/diagnostic/create_ref_usage.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/diagnostic_test_base.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(CreateRefUsageDiagnosticTest);
+  });
+}
+
+@reflectiveTest
+class CreateRefUsageDiagnosticTest extends DiagnosticTestBase {
+  @override
+  get errorUnderTest => CreateRefUsageDiagnostic.code;
+
+  @override
+  get fixKindUnderTest => CreateRefUsageDiagnostic.fixKind;
+
+  Future<void> test_noErrorUseRef() async {
+    final source = newSource('test.dart', /*language=dart*/ r'''
+import 'package:over_react/over_react.dart' hide createRef;
+
+part 'test.over_react.g.dart';
+
+mixin FooProps on UiProps {}
+
+final Foo = uiFunction<FooProps>(
+  (props) {
+    final ref = useRef();
+    
+    return (Dom.div()..ref = ref)();
+  },
+  _$FooConfig, // ignore: undefined_identifier
+);
+''');
+    expect(await getAllErrors(source), isEmpty);
+  }
+
+  Future<void> test_noErrorNotFromOverReact() async {
+    final source = newSource('test.dart', /*language=dart*/ r'''
+import 'package:over_react/over_react.dart' hide createRef;
+
+part 'test.over_react.g.dart';
+
+mixin FooProps on UiProps {}
+
+final Foo = uiFunction<FooProps>(
+  (props) {
+    String createRef() => 'abc';
+    
+    return Dom.div()(createRef());
+  },
+  _$FooConfig, // ignore: undefined_identifier
+);
+''');
+    expect(await getAllErrors(source), isEmpty);
+  }
+
+  /// Helper function to expect that the `createRef` usage error exists and verify that
+  /// the fix will replace it with `useRef`.
+  Future<void> _expectErrorAndFix(String input, String expectedOutput) async {
+    var source = newSource('test.dart', input);
+    final selection = createSelection(source, "#createRef#");
+
+    // Verify error.
+    expect((await getAllErrors(source)).single, isAnErrorUnderTest(locatedAt: selection, hasFix: true));
```

Comment:
#nit I think you can use `expectSingleErrorAt` here instead:
```suggestion
    await expectSingleErrorAt(selection);
```